spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binder/AbstractBinder.java

```diff
@@ -151,22 +146,22 @@ protected final String groupedName(String name, String group) {
 		return name + GROUP_INDEX_DELIMITER + (StringUtils.hasText(group) ? group : "default");
 	}
 
+	@Deprecated
 	protected final MessageValues serializePayloadIfNecessary(Message<?> message) {
```

Comment:
Shouldn't we just add these deprecations in 1.3 and remove them in 2.0? Otherwise we can't remove them until 3.0.

Reply:
But I guess that does not have to happen before the merge.

Reply:
Well, we can't deprecated in `1.3` since it is a first class citizen in the logic there.
I'm fine stick with deprecation until `3.0` as far as this method will be out of use any way.